### PR TITLE
WD-3159 - Add some API tests

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Added tests to some of the API methods.
- Added test factories and moved ModelManager factories to their own file.

## Details

https://warthogs.atlassian.net/browse/WD-3159

### CODE REVIEW

File: src/juju/api.ts [hunk 1]
```diff
@@ -232,17 +232,12 @@ export async function fetchModelStatus(
...
     try {
       const controllerCredentials = getUserPass(getState(), wsControllerURL);
-      const response = await connectAndLoginWithTimeout(
+      const { conn, logout } = await connectAndLoginWithTimeout(
         modelURL,
         controllerCredentials,
         generateConnectionOptions(false),
         useIdentityProvider
       );
-      if (typeof response === "string") {
```
Comment (huwshimi) [inline src/juju/api.ts:241]: Once I corrected the type it revealed this wasn't actually possible.

File: src/store/juju/selectors.test.ts [hunk 2]
```diff
@@ -1,49 +1,48 @@
 import { rootStateFactory } from "testing/factories";
 import { generalStateFactory } from "testing/factories/general";
+import { modelStatusInfoFactory } from "testing/factories/juju/Clientv6";
+import { modelUserInfoFactory } from "testing/factories/juju/ModelManagerV9";
 import {
-  jujuStateFactory,
```
Comment (huwshimi) [inline src/store/juju/selectors.test.ts:4]: I guess my IDE alphabetised these imports.

File: src/testing/factories/juju/ModelManagerV9.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import type {
```
Comment (huwshimi) [inline src/testing/factories/juju/ModelManagerV9.ts:1]: These factories were moved from juju.ts.

**Review by Lukewh — APPROVED** [review]
:+1: LGTM

**Pull request merged**